src/python/director/startup.py

```diff
@@ -685,38 +686,45 @@ def onPlanStatus(msg):
         if polyData:
             vis.showPolyData(polyData, os.path.basename(filename))
 
-
-if useImageWidget:
-    imageWidget = cameraview.ImageWidget(cameraview.imageManager, 'CAMERA_LEFT', view)
-    #imageWidget = cameraview.ImageWidget(cameraview.imageManager, 'KINECT_RGB', view)
-
 if useCameraFrustumVisualizer and cameraview.CameraFrustumVisualizer.isCompatibleWithConfig():
     cameraFrustumVisualizer = cameraview.CameraFrustumVisualizer(robotStateModel, cameraview.imageManager, 'CAMERA_LEFT')
 
 class ImageOverlayManager(object):
 
     def __init__(self):
         self.viewName = 'CAMERA_LEFT'
-        #self.viewName = 'KINECT_RGB'
-        self.size = 400
+        self.desiredWidth = 400
+        self.size = [self.desiredWidth, self.desiredWidth/(4/3.)]
         self.position = [0, 0]
         self.usePicker = False
         self.imageView = None
         self.imagePicker = None
         self._prevParent = None
+        self.imageSize = [640, 480]
+        self.imageAspectRatio = 4/3.
 
-    def show(self):
+    def setWidth(self, width):
+        self.desiredWidth = width
+        self.hide()
+        self.show()
 
+    def show(self):
         if self.imageView:
             return
 
         imageView = cameraview.views[self.viewName]
         self.imageView = imageView
         self._prevParent = imageView.view.parent()
 
+        imageExtent = cameraview.imageManager.images[self.viewName].GetExtent()
+        if imageExtent[1] != -1 and imageExtent[3] != -1:
+            self.imageSize = [imageExtent[1]+1, imageExtent[3]+1]
+            self.imageAspectRatio = self.imageSize[0] / self.imageSize[1]
```

Comment:
oh i see it is set here.  this is not the best place to set it because this code is only called the first time the overlay is shown, which might be before the image exists.

imageAspectRatio should not need to be a stored variable, it's an attribute of the image data